src/keboola/component/dao.py

```diff
@@ -435,16 +525,66 @@ def _filter_attributes_by_manifest_type(self, manifest_type: Literal["in", "out"
         Returns:
 
         """
-        supported_fields = self._manifest_attributes.get_attributes_by_stage(manifest_type, legacy_queue)
 
-        new_dict = self._raw_manifest.copy()
-        if supported_fields:
-            for attr in self._raw_manifest:
-                if attr not in supported_fields:
-                    new_dict.pop(attr, None)
-        return new_dict
+        if isinstance(self, TableDefinition):
+            supported_fields = self._manifest_attributes.get_attributes_by_stage(manifest_type, legacy_queue,
+                                                                                 native_types)
+            fields = {
+                # TODO: add input manifest attributes
+                # 'id': None,
+                # 'uri': None,
+                # 'name': self.name,
+                # 'created': None,
+                # 'last_change_date': None,
+                # 'last_import_date': None,
+                'destination': self.destination,
+                'columns': self.columns if native_types else self.legacy_columns,
+                'incremental': self.incremental,
+                'primary_key': self.primary_key,
+                'write_always': self.write_always,
+                'delimiter': self.delimiter,
+                'enclosure': self.enclosure,
+                'metadata': self.table_metadata.get_table_metadata_for_manifest() if self.table_metadata else None,
+                'column_metadata': self.table_metadata.get_column_metadata_for_manifest() if self.table_metadata else None, # noqa
+                'manifest_type': manifest_type,
+                'has_header': self._has_header_in_file(manifest_type),
+                'description': None,
+                'table_metadata': None,
+                'delete_where_column': self.delete_where_column,
+                'delete_where_values': self.delete_where_values,
+                'delete_where_operator': self.delete_where_operator,
+                'schema': [col.to_dict() for col in self.schema] if self.schema else []
+            }
+
+            new_dict = fields.copy()
+
+            if supported_fields:
+                for attr in fields:
+                    if attr not in supported_fields:
+                        new_dict.pop(attr, None)
+            return new_dict
 
-    def get_manifest_dictionary(self, manifest_type: Optional[str] = None, legacy_queue=False) -> dict:
+        else:
+            return {
+                "tags": self.tags,
+                "is_public": self.is_public,
+                "is_permanent": self.is_permanent,
+                "is_encrypted": self.is_encrypted,
+                "notify": self.notify
+            }
+
+    def _has_header_in_file(self, manifest_type):
```

Comment:
manifest_type je v `self.stage`. Pro zachovani stavajici konvence, tohle by mela byt property a kdyz se rozhodneme zachovat _raw_manisfest. tak taky updatovat manifest reprezentaci.

Tohle je neco co muze nastavovat user ne? To je cela pointa toho to mit v manifestu. Tady je potreba zjistit jestli plati tohle https://keboola.atlassian.net/wiki/spaces/ENGG/pages/3099787265?focusedCommentId=3232432235 (doufam ze ne! )

Reply:
Takže to, co psal Odin, platí? Btw, to první has_header nebude nikdy použitý.